src/utils/validation.ts

```diff
@@ -23,6 +23,14 @@ export function checkArray (array: Tensor | RecursiveArray<number>, dtype = 'any
   });
 }
 
+
+export function checkJSArray (array: Tensor | RecursiveArray<number>, dtype = 'any', ensureDimension = -1): number | number[] | number[][] | number[][][] | number[][][][] | number[][][][][] | number[][][][][][] {
+  return tidy(() => {
+    const tensor = checkArray(array, dtype, ensureDimension);
```

Comment:
Why does the `checkArray()` function return a tensor? It sounds like a validation function and it should return a boolean value IMO.